backend/src/main/java/com/redhat/cloud/notifications/models/KafkaMessageId.java

```diff
@@ -0,0 +1,39 @@
+package com.redhat.cloud.notifications.models;
+
+import javax.persistence.Embeddable;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.UUID;
+
+@Embeddable
+public class KafkaMessageId implements Serializable {
+
+    @NotNull
+    public String bundleName;
+
+    @NotNull
+    public String applicationName;
+
+    @NotNull
+    public UUID messageId;
```

Comment:
I'm still not sure about the need to store the bundle/app in the database.

If we do that, then we can delegate the responsibility of the `messageId` unicity to the onboarded apps. Duplicates would be their responsibility, not ours.

On the other hand, the chance for a UUID V4 collision is so low that it's very unlikely. Furthermore, the `kafka_message` table is meant to be purged regularly because there's no point storing old messages IDs forever. This reduces even more the collision risk.

Version 4 of UUID is not enforced right now though. They could send us a V3 or lower and it would still be valid for `UUID.fromString`. Maybe we should enforce V4 UUID in the header and get rid of bundle/app in the DB because of the super low collision risk.

WDYT @josejulio?

Reply:
cc @pilhuhn

Reply:
You will probably need to have a look at `KafkaMessageDeduplicator` to understand the question since the PR description is still missing. 😄

Reply:
It would make sense to use only V4... but

This github issue scared me a bit: https://github.com/ramsey/uuid/issues/80
TLDR; There is/was a bug on openssl that increases the chance of collision (if the generator uses a particular function of openssl) when using forked process (collisions across the forked)

Anyhow, the collisions would be for a single application. 
I'm +1 on removing the app/bundle and enforce v4.

Reply:
I wonder if we should store literally as done here, as opposed to use a FK to the apps table, which has a FK to the bundle.

Tbh. what is the collision risk even with that uuid/80 issue? a few hundreds on a million tries? That should be good enough (and the issue is 3-5y old, so most likely fixed)

I think just storing uuid + timestamp could be enough. TS so that we can purge entries older than e.g. a day (I guess the value could be lower)

Reply:
I don't think we need the relationship between `KafkaMessage` and the app in the database. I added the `bundle`/`application` columns only for unicity purposes but since we all agree that they can be removed, I'll get rid of them. Thanks for your opinion on this @josejulio @pilhuhn!

Reply:
Done.